composer/core/data_spec.py

```diff
@@ -295,6 +298,23 @@ def _default_get_num_tokens_in_batch(self, batch: Batch) -> int:
             return self.dataloader.dataset.max_seq_len * samples_per_batch  # type: ignore
         return 0
 
+    def get_num_tokens_in_batch(self, batch: Batch, token_type: str = 'total') -> int:
+        num_tokens = self._get_num_tokens_in_batch(batch)
+
+        if isinstance(num_tokens, int):
+            if token_type != 'total':
+                log.warning(
+                    f'get_num_tokens_in_batch returned an int, but token_type is {token_type}. ' +
+                    'Returning the total number of tokens in the batch.',
```

Comment:
```suggestion
            if token_type != 'total':
                log.warning(
                    f'get_num_tokens_in_batch returned an int, but token_type was specified as {token_type}. ' +
                    'Returning the total number of tokens in the batch.',
```
nit: make more clear that this was user set